checker/src/org/checkerframework/checker/genericeffects/GenericEffectTypeFactory.java

```diff
@@ -71,6 +77,96 @@ public ExecutableElement findJavaOverride(ExecutableElement overrider, TypeMirro
     }
 
     /**
+     * This method is used to get the inner most class with a DefaultEffect annotation from an element (in case there are nested classes).
+     * Note: If the developer would not like this feature. The commented code at the bottom will only return the enclosing class of the element
+     * and ignore nested classes.
+     *
+     * @param elt An element for which a DefaultEffect annotated class has to be found.
+     * @return Inner most annotated class with the DefaultEffect annotation. If none is found then the outer most class is returned.
+     */
+    private Element getInnermostAnnotatedClass(Element elt) {
+        Element clsElt = elt;
+        while (clsElt != null) {
+            if (clsElt.getAnnotation(DefaultEffect.class) != null)
+                break;
+            else
+                clsElt = clsElt.getEnclosingElement();
```

Comment:
Technically this could be a while loop, since it's often the case that an anonymous inner class is declared inside a method, which would be the anonymous class's enclosing element.  This works just fine, but please put some indication here that ```clsElt``` will sometimes hold elements other than class elements.  A comment is fine, or a loop that continues retrieving enclosing elements until a class or interface is encountered.